### PR TITLE
Milestone 3: Game logic

### PR DESCRIPTION
**Pull request by Misori-simon**
- we build a player class to handle all the methods and properties of a player
- we made a game class to handle all the logical interactions of the game and the players
- we linked our user interface to the logic established a cohesive interaction for the palyer.

### CODE REVIEW

File: lib/logic.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    if move_set.length >= 3
+      @winning_cases.each do |item|
+        return true if item.subset?(move_set)
+      end
+    end
+    false
+  end
+
+  def current_move_getter
+    @current_move = gets.chomp.strip.downcase
```
Comment (PhillipUg) [inline lib/logic.rb]: - [x] No `gets.chomp` should be used outside the `main.rb` file.

File: lib/logic.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  def draw_board
+    # Draws the game board to the concole
+    puts "\n\n  -------------------"
+    puts "A | #{@board_values['a1']} | #{@board_values['a2']} | #{@board_values['a3']} |"
+    puts '  -------------------'
+    puts "B | #{@board_values['b1']} | #{@board_values['b2']} | #{@board_values['b3']} |"
+    puts '  -------------------'
+    puts "C | #{@board_values['c1']} | #{@board_values['c2']} | #{@board_values['c3']} |"
+    puts '  -------------------'
+    puts "    1     2    3    \n\n"
```
Comment (PhillipUg) [inline lib/logic.rb]: - [x] No `puts/print` should be used outside the `main.rb` file

File: lib/logic.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+  def set_turn
+    # Increments @turn in order to pass turn to other player
+    @turn += 1
+  end
+
+  def make_move
+    # Makes a move for current_player
+    puts "It's #{current_player.name}'s turn"
+    print 'Enter your move, e.g (A1, B3) => '
```
Comment (PhillipUg) [inline lib/logic.rb]: - [x] remove this `puts/print` from here as well

File: lib/logic.rb [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    @board_values.each do |position|
+      if value == position[0]
+        @board_values[position[0].to_s] = @turn.odd? ? "\sX\s" : "\sO\s"
+      end
+    end
+  end
+
+  def valid_moves
+    # Constructs an array that holds valid/allowed moves
+    # based on cells of @grid matrix
```
Comment (PhillipUg) [inline lib/logic.rb]: - [x] remove inline comments

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey **Cy Kalu & Simon Misori**, 👋

You did an amazing job on this project. The logic works well.

However, there are just a few project requirements that you missed, so you need to fix them before you can go on to the next milestone.

### Changes Required ♻️ 

- [x] All `puts/print` and `gets.chomp` statements should be moved to the `main.rb` file. 

- [x] Remove all inline comments from the previous milestone. Leave just the logic

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**

I would be happy to help! 👩‍💻 

Happy coding 🤓

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @cyonii and @Misori-simon !

### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:

### You may move on to the next milestone, good luck with your next projects! :rocket:
### Happy Coding! :+1:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Pull request merged**